cheroot/server.py

```diff
@@ -1792,19 +1792,20 @@ def prepare(self):  # noqa: C901  # FIXME
 
     def serve(self):
         """Serve requests, after invoking :func:`prepare()`."""
-        self.serving = True
-        while self.ready:
-            try:
-                self.tick()
-            except (KeyboardInterrupt, SystemExit):
-                raise
-            except Exception:
-                self.error_log(
-                    'Error in HTTPServer.tick', level=logging.ERROR,
-                    traceback=True,
-                )
-
-        self.serving = False
+        try:
+            self.serving = True
+            while self.ready:
+                try:
+                    self.tick()
+                except (KeyboardInterrupt, SystemExit):
+                    raise
```

Comment:
It's cleaner to set `self.serving = False` here instead of having a huge try/finally with unnecessary nesting.

Reply:
The advantage of the `try`-`finally` block is that it's future-proofed against other `except` blocks being added and reintroducing the bug.

Reply:
True, but readability matters. Maybe it could be moved to a decorator or a context manager, then, so that it doesn't contribute to nesting.

As for future bugs, it's good that you reminded me: this change has to be accompanied by a regression test in order to be accepted.

Reply:
Done.